lib/std/os/linux.zig

```diff
@@ -874,6 +874,11 @@ pub const MSF = struct {
     pub const SYNC = 4;
 };
 
+/// Can only be called on 64 bit systems.
+pub fn mseal(address: [*]const u8, length: usize, flags: u64) usize {
```

Comment:
```suggestion
pub fn mseal(address: [*]const u8, length: usize, flags: c_ulong) usize {
```

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/diff/Documentation/userspace-api/mseal.rst?id=0b32d436c015d5a88b3368405e3d8fe82f195a54

Reply:
Maybe I'm missing something, but there is no `c_ulong` in the syscall wrappers.
For example, the syscall `map_shadow_stack` uses two `unsigned long` ([Linux source](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/x86/kernel/shstk.c#n505)) which are u64 in the [wrapper](https://github.com/ziglang/zig/blob/531cd177e89c1edfcd2e52f74f220eb186a25f78/lib/std/os/linux.zig#L2375).

Reply:
Yeah, we've been bad about this historically. We also mix up `fd_t`, `i32`, and `u32`, to give another example. I would say that, in general, it is ok to substitute ABI-equivalent types when they add safety/clarity (e.g. `fd_t` instead of `i32`, pointers instead of `c_ulong`, etc). But beyond that, deviating from the kernel signature hurts clarity rather than helping it, because it forces someone auditing the code to wonder why the decision to deviate was made (when there likely is no particularly compelling reason).

If this system call ever becomes available on 32-bit (unlikely, but not impossible), we would have to remember to go fix the signature to use `c_ulong` instead of `u64`. Better to just do it right from the beginning.

Reply:
That makes sense. I will fix it.